openassessment/xblock/ui_mixins/mfe/mixin.py

```diff
@@ -30,3 +49,83 @@ def get_block_learner_assessment_data(self, data, suffix=""):  # pylint: disable
         serializer_context = {"view": "assessment"}
         page_context = PageDataSerializer(self, context=serializer_context)
         return page_context.data
+
+    def _submission_draft(self, data):
+        try:
+            student_submission_data = data['response']['text_responses']
+            submissions_actions.save_submission_draft(student_submission_data, self.config_data, self.submission_data)
+        except KeyError as e:
+            raise OraApiException(400, ErrorCodes.INCORRECT_PARAMETERS) from e
+        except SubmissionValidationException as exc:
+            raise OraApiException(400, ErrorCodes.INVALID_RESPONSE_SHAPE, str(exc)) from exc
+        except DraftSaveException as e:
+            raise OraApiException(500, ErrorCodes.INTERNAL_EXCEPTION) from e
+
+    def _submission_create(self, data):
+        from submissions import api as submission_api
```

Comment:
Nit: more descriptive names would be helpful here for readability

```suggestion
    def _save_draft_response_and_wrap_error(self, data):
        try:
            student_submission_data = data['response']['text_responses']
            submissions_actions.save_submission_draft(student_submission_data, self.config_data, self.submission_data)
        except KeyError as e:
            raise OraApiException(400, ErrorCodes.INCORRECT_PARAMETERS) from e
        except SubmissionValidationException as exc:
            raise OraApiException(400, ErrorCodes.INVALID_RESPONSE_SHAPE, str(exc)) from exc
        except DraftSaveException as e:
            raise OraApiException(500, ErrorCodes.INTERNAL_EXCEPTION) from e

    def _submit_response_and_wrap_error(self, data):
        from submissions import api as submission_api
```

Reply:
I don't think that error wrapping is really something that needs to be specified in these method names, but I will try to make these names more descriptive